core/solver/gmres_kernels.hpp

```diff
@@ -38,11 +38,19 @@ namespace gmres {
                     stopping_status* stop_status)
 
 
-#define GKO_DECLARE_ALL_AS_TEMPLATES             \
-    template <typename ValueType>                \
-    GKO_DECLARE_GMRES_RESTART_KERNEL(ValueType); \
-    template <typename ValueType>                \
-    GKO_DECLARE_GMRES_MULTI_AXPY_KERNEL(ValueType)
+#define GKO_DECLARE_GMRES_MULTI_DOT_KERNEL(_type)               \
+    void multi_dot(std::shared_ptr<const DefaultExecutor> exec, \
+                   const matrix::Dense<_type>* krylov_bases,    \
+                   const matrix::Dense<_type>* next_krylov,     \
+                   matrix::Dense<_type>* hessenberg_col)
```

Comment:
nit: maybe `hessenberg_transposed_col` would be a better name, but I get that it might be too long

Reply:
What about `hess_trans_col` or something like that? Unless we change the layout of `hessenberg_iter`, in which case `hessenberg_col` would be correct after all...

Reply:
I guess this just up to personal preference. I would be ok with that.

Reply:
Since `multi_dot` works with `hessenberg_iter` which is now viewed as columns again, I left the name for now.